libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/roomlist/RoomSummaryListProcessor.kt

```diff
@@ -120,14 +118,20 @@ class RoomSummaryListProcessor(
 
     private suspend fun buildAndCacheRoomSummaryForIdentifier(identifier: String): RoomSummary {
         val builtRoomSummary = roomListService.roomOrNull(identifier)?.use { roomListItem ->
-            RoomSummary.Filled(
-                details = roomSummaryDetailsFactory.create(roomListItem)
-            )
+            buildAndCacheRoomSummaryForRoomListItem(roomListItem)
         } ?: buildEmptyRoomSummary()
         roomSummariesByIdentifier[builtRoomSummary.identifier()] = builtRoomSummary
         return builtRoomSummary
     }
 
+    private suspend fun buildAndCacheRoomSummaryForRoomListItem(roomListItem: RoomListItem): RoomSummary {
+        val builtRoomSummary = RoomSummary.Filled(
```

Comment:
We don't need to keep Empty and Filled, right?

Reply:
Actually, we might still need the `Empty` case it if we try to create an item from the identified and the `RoomListService` can't find it.

Reply:
Actually, it *seems* like this is possible. I'm a bit worried about what could happen to the cache if no `RoomListItem` can be retrieved though 🫤 .